lib/src/logic/questionnaire_controller.dart

```diff
@@ -85,109 +95,127 @@ class QuestionnaireController {
-          case QuestionnaireItemType.decimal:
-            itemView = QuestionnaireDecimalItemView(
-              item: item,
-              enableWhenController: enableWhenController,
-            );
-            break;
-          case QuestionnaireItemType.boolean:
-            itemView = QuestionnaireBooleanItemView(
-              item: item,
-              enableWhenController: enableWhenController,
-            );
-            break;
-          case QuestionnaireItemType.choice:
-            itemView = buildChoiceItemView(
-                item: item, enableWhenController: enableWhenController);
-            break;
-          case QuestionnaireItemType.openChoice:
-            itemView = buildOpenChoiceItemView(
-                item: item, enableWhenController: enableWhenController);
-            break;
-          case QuestionnaireItemType.date:
-          case QuestionnaireItemType.time:
-          case QuestionnaireItemType.dateTime:
-            itemView = QuestionnaireDateTimeItemView(
-              item: item,
-              enableWhenController: enableWhenController,
-              type: DateTimeType.fromQuestionnaireItemType(itemType),
-            );
-            break;
-          case QuestionnaireItemType.quantity:
-            itemView = QuestionnaireQuantityItemView(
-              item: item,
-              enableWhenController: enableWhenController,
-            );
-            break;
-          case QuestionnaireItemType.url:
-            itemView = QuestionnaireUrlItemView(
-              item: item,
-              enableWhenController: enableWhenController,
-            );
-            break;
-          case QuestionnaireItemType.display:
-            itemView = QuestionnaireDisplayItemView(
-              item: item,
-              enableWhenController: enableWhenController,
-            );
-            break;
-          case QuestionnaireItemType.attachment:
-            itemView = QuestionnaireAttachmentItemView(
-              item: item,
-              onAttachmentLoaded: onAttachmentLoaded,
-              enableWhenController: enableWhenController,
-            );
-            break;
-          case QuestionnaireItemType.group:
-            children = buildQuestionnaireItemBundles(item.item,
-                onAttachmentLoaded: onAttachmentLoaded);
-            itemView = QuestionnaireGroupItemView(
-              item: item,
-              enableWhenController: enableWhenController,
-              children: children.map((itemBundle) => itemBundle.view).toList(),
-            );
-            break;
-          default:
-        }
-        if (itemView != null) {
+        if (customItemView != null) {
           itemBundles.add(QuestionnaireItemBundle(
             item: item,
-            view: itemView,
-            children: children,
-            controller: itemView.controller,
+            view: customItemView,
+            controller: customItemView.controller,
           ));
+        } else {
```

Comment:
The problem I see here with the override of itemView is that the `QuestionnaireItemBundle` would lack of children when the itemType is `group`.

Reply:
Maybe the override should be to the whole `QuestionnaireItemBundle`...